src/lf_mask.rs

```diff
@@ -157,14 +180,17 @@ fn mask_edges_inter(
     let t_dim = &dav1d_txfm_dimensions[max_tx as usize];
 
     // See [`decomp_tx`]'s docs for the `txa` arg.
-    let mut txa = Align16([[[[0; 32]; 32]; 2]; 2]);
+    let mut txa: MaybeUninit<Align16<[[[[u8; 32]; 32]; 2]; 2]>> = MaybeUninit::uninit();
 
     for (y_off, _) in (0..h4).step_by(t_dim.h as usize).enumerate() {
         for (x_off, _) in (0..w4).step_by(t_dim.w as usize).enumerate() {
-            decomp_tx(&mut txa.0, max_tx, 0, y_off as u8, x_off as u8, tx_masks);
+            decomp_tx(&mut txa, max_tx, 0, y_off as u8, x_off as u8, tx_masks);
         }
     }
 
+    // SAFETY: Calls to `decomp_tx` above fully initializes the `txa` array
```

Comment:
We need a more comprehensive safety comment here.  It's not trivial at all why all of the `decomp_tx` calls fully initializes `txa`.

Reply:
I'd also add a `# Safety` section to `decomp_tx` saying what part of `txa` it initializes so that it is easier to reason about here.

Reply:
If the C code got this wrong we'd see random test failures. You're right, but I'm not sure I want to spend the time right now to reason about code that I don't fully understand but that is definitely correct. These case_sets and the entire logic here isn't clear.

Reply:
The major advantage of `rav1d` over `dav1d` is safety, so I think we should be rigorous about safety like this where it's reasonable (so not counting asm calls and `DisjointMut`).  Random test failures wouldn't be guaranteed I think, as `txa` could be zeroed anyways.

Reply:
Turns out it wasn't fully initialized, there are some gaps in the `step` dimension (only block starting points are populated). I've thoroughly commented it all and we now keep the array `MaybeUninit` the whole lifetime.